### PR TITLE
refactor(event_handler): add from __future__ import annotations in the Middlewares

### LINKED ISSUE #4982: Tech debt: add `from __future__ import annotations` in the Event Handler Middlewares feature
### Why is this needed?

Reference: https://github.com/aws-powertools/powertools-lambda-python/issues/4607

Add from future import annotations to all files, including examples

### Which area does this relate to?

Static typing

### Suggestion

_No response_

### Acknowledgment

- [X] This request meets [Powertools for AWS Lambda (Python) Tenets](https://docs.powertools.aws.dev/lambda/python/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Java](https://github.com/aws-powertools/powertools-lambda-java/), [TypeScript](https://github.com/aws-powertools/powertools-lambda-typescript/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### PR DESCRIPTION
**Pull request by ericbn**
**Issue number:** #4982

### Changes

Add `from __future__ import annotations` to middlewares package

### User experience

Discussed in #4607

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [X] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [X] I have performed a self-review of this change
* [X] Changes have been tested
* [ ] Changes are documented
* [X] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: aws_lambda_powertools/event_handler/middlewares/openapi_validation.py [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-from typing import Any, Callable, Dict, List, Mapping, MutableMapping, Optional, Sequence, Tuple
+from typing import TYPE_CHECKING, Any, Callable, Mapping, MutableMapping, Sequence
 
 from pydantic import BaseModel
 
-from aws_lambda_powertools.event_handler import Response
-from aws_lambda_powertools.event_handler.api_gateway import Route
 from aws_lambda_powertools.event_handler.middlewares import BaseMiddlewareHandler, NextMiddleware
 from aws_lambda_powertools.event_handler.openapi.compat import (
     ModelField,
```
Comment (leandrodamascena) [inline aws_lambda_powertools/event_handler/middlewares/openapi_validation.py]: Should `ModelField` be imported in TYPE_CHECKING section? Same with `NextMiddleware`?

**Review by leandrodamascena — CHANGES_REQUESTED** [review]
Hey @ericbn! I left only one small comment before we merge this.

File: aws_lambda_powertools/event_handler/middlewares/openapi_validation.py (hunk 1 above)
Comment (ericbn) [reply aws_lambda_powertools/event_handler/middlewares/openapi_validation.py]: Pydantic [does work](https://docs.pydantic.dev/latest/concepts/postponed_annotations/) with `from __future__ import annotations`, but moving some imports into TYPE_CHECKING would completely disable them in the [normal (outside type-checkers) runtime](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING). Any type annotation used by pydantic must be out of TYPE_CHECKING.

Comment (ericbn) [reply aws_lambda_powertools/event_handler/middlewares/openapi_validation.py]: Oh, but looks like you're right, these are not being used in a pydantic model in this code. I was relying mostly on ruff to detect what needs to go to TYPE_CHECKING, and it seems to have missed these.

**Review by leandrodamascena — APPROVED** [review]
APPROVED!

**Pull request merged**